modules/client/src/tokenVoting/internal/client/methods.ts

```diff
@@ -333,41 +332,39 @@ export class TokenVotingClientMethods extends ClientCore
     if (!isProposalId(proposalId)) {
       throw new InvalidProposalIdError();
     }
-    try {
-      await this.graphql.ensureOnline();
-      const client = this.graphql.getClient();
-      const extendedProposalId = getExtendedProposalId(proposalId);
-      const {
-        tokenVotingProposal,
-      }: {
-        tokenVotingProposal: SubgraphTokenVotingProposal;
-      } = await client.request(QueryTokenVotingProposal, {
+    const extendedProposalId = getExtendedProposalId(proposalId);
+    const {
+      tokenVotingProposal,
+    }: {
+      tokenVotingProposal: SubgraphTokenVotingProposal;
+    } = await this.graphql.request({
+      query: QueryTokenVotingProposal,
+      params: {
         proposalId: extendedProposalId,
-      });
-      if (!tokenVotingProposal) {
-        return null;
-      }
-      // format in the metadata field
-      try {
-        const metadataCid = resolveIpfsCid(tokenVotingProposal.metadata);
-        const metadataString = await this.ipfs.fetchString(metadataCid);
-        const metadata = JSON.parse(metadataString) as ProposalMetadata;
-        return toTokenVotingProposal(tokenVotingProposal, metadata);
-        // TODO: Parse and validate schema
-      } catch (err) {
-        if (err instanceof InvalidCidError) {
-          return toTokenVotingProposal(
-            tokenVotingProposal,
-            UNSUPPORTED_PROPOSAL_METADATA_LINK,
-          );
-        }
+      },
+      name: "TokenVoting proposal",
+    });
+    if (!tokenVotingProposal) {
+      return null;
+    }
+    // format in the metadata field
+    try {
```

Comment:
Why does the `try` block only cover half of the function code that could fail?

Reply:
Because it handles the ipfs errors, this will eventually be handled like the subgraph ones are handled on this PR